src/database/utilities/createRecord.js

```diff
@@ -82,63 +130,134 @@ const createAddress = (database, { line1, line2, line3, line4, zipCode } = {}) =
 /**
  * Creates a prescriber record. prescriberDetails can have the shape:
  * {
- *     firstName, lastName, registrationCode, line1, line2, isVisible,
+ *     id, firstName, lastName, registrationCode, address1, address2, isVisible,
  *     isActive, phoneNumber, mobileNumber, emailAddress
  * }
  */
 const createPrescriber = (database, prescriberDetails) => {
-  const { addressOne, addressTwo } = prescriberDetails;
-  const address = createAddress(database, { line1: addressOne, line2: addressTwo });
+  const {
+    id: prescriberId,
+    firstName,
+    lastName,
+    registrationCode,
+    address1,
+    address2,
+    phoneNumber,
+    mobileNumber,
+    emailAddress,
+    storeId: prescriberStoreId,
+  } = prescriberDetails;
+
+  const id = prescriberId ?? generateUUID();
+  const thisStoreId = database.getSetting(SETTINGS_KEYS.THIS_STORE_ID)
+  const supplyingStoreId = prescriberStoreId ?? thisStoreId;
+  const fromThisStore = supplyingStoreId === thisStoreId;
+
+  const address = getOrCreateAddress(database, address1, address2);
+
+  const isVisible = true;
+  const isActive = true;
```

Comment:
visibility is a given but is the `isActive` 100%? Possible to get an `active = false`?

Reply:
Hmm, yeah, wasn't sure about this one! I think maybe set to given value or `true` if `undefined`?

Reply:
80002c59502022f4737374158f1f38db42bd7b15.